### ISSUE TITLE
Integrate with XBEL bookmarks database

### ISSUE THREAD

**Issue by GreenLunar** [yacy/yacy_search_server#452]
e: thread starts [here](https://github.com/yacy/yacy_search_server/issues/452#issuecomment-1031091385)

~[buku](https://github.com/jarun/buku) is a powerful bookmark manager and a personal textual mini-web.~

~I want to share my bookmarks privately, using access token, with co-workers and friends.
It would be nice to add support to retrieve results from a buku `bookmarks.db` database file.~

Related: https://github.com/jarun/buku/issues/549

**Comment by reger24** [comment]
don't see how that could be realized in yacy_search_server.
buku depends on python and yacy_search_server is pure java - and adding a dependency to python seems not desirable for all the different systems (specially not on Win)

**Comment by GreenLunar** [comment]
It's just the `bookmarks.db` database which is sqlite3. Python is *not* required.

**Comment by Dalethium** [comment]
I think, if this were to be integrated, it may be more fruitful to use a more widely adopted external bookmark system such as [XBEL](https://en.wikipedia.org/wiki/XBEL) -- that way more widely supported systems such as [Flocus](https://floccus.org/start#3-choose-a-sync-method) could be used along with many others. 

I think that using something like XBEL is smarter since it's a format standard, and not a specific project that Yacy would have to then integrate and work with. That way, if other projects wanted to integrate with Yacy's bookmark system, they could follow a standard format and not depend on this project or any other project for integration practices.

**Comment by GreenLunar** [comment]
> 


![image](https://user-images.githubusercontent.com/2053884/152735087-586b819b-b26f-447c-b3e1-ad90d5fbfc68.png) https://floccus.org/assets/img/screen_select_sync_method.png

**Comment by reger24** [comment]
fyi:   bookmarks can be exported in xbel   see  http://localhost:8090/api/bookmarks/xbel/xbel.xml

**Comment by GreenLunar** [comment]
Is it possible to make them appear in in search results too?
I'm asking because I'm not familiar with YaCy.

**Comment by reger24** [comment]
Every search result (which was not already bookmarked by user) has a bookmark icon
![image](https://user-images.githubusercontent.com/11603289/153072299-2c0bab1f-b901-41b7-90c5-69165b855566.png)

and can be viewed etc. on a separate page
![image](https://user-images.githubusercontent.com/11603289/153073299-73a73a57-7ab4-424e-ba63-90a7219fdfec.png)

YaCy also creates automatic bookmarks for crawl start pages (this can be made public and shared with other peers)

P.S. currently not all parts of the bookmark feature is activated (@Orbiter : maybe it's time to decide if we want to further support the bookmark feature - after the ymark stuff is removed - -> if not, we should delete all bmk source, or activate the remains).

**Comment by GreenLunar** [comment]
Is there a dicumentation about this?

**Comment by reger24** [comment]
no, I don't think a documentation exists (only the user interface hints),

(I know only the documentation in the old [Wiki](https://wiki.yacy.net/index.php/En:Start) )
